### PR TITLE
Add FsEmployee GET jwt token endpoint

### PR DESCRIPTION
**Pull request by bruce-fs**
Update Request interface to handle GET along with POST requests
Update README and gemspec

### CODE REVIEW

File: lib/hubble_observatory/fs_employee.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module HubbleObservatory
+# Provides methods to interact with the Hubble API fs-employees endpoint
+# @see https://hubble.fullscreen.net/api/docs#fsemployees-getting-jwt-token
+  class FsEmployee
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/fs_employee.rb]: @bruce-fs Should this be `create`? (According to the README)

Comment (claudiofullscreen) [inline lib/hubble_observatory/fs_employee.rb]: @bruce-fs Should this be `Employee`? (According to the README)

File: lib/hubble_observatory/request.rb [hunk 2]
```diff
@@ -12,16 +12,18 @@ class Request
...
 
     # Sends the request and returns the response
     def run_request
-      net_http_class = Object.const_get("Net::HTTP::#{@request_type.capitalize}")
-      request = net_http_class.new uri
-      request.body = serialize_attributes(attributes: @body_attrs).to_json
-      response_for(assign_headers(request), uri)
+      if response.is_a?(Net::HTTPSuccess) || response.is_a?(Net::HTTPUnprocessableEntity)
+        parse(response)
+      else
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs This `else nil` is unnecessary

File: lib/hubble_observatory/request.rb [hunk 3]
```diff
@@ -41,32 +43,62 @@ def serialize_attributes(attributes:)
...
+
+    def assign_request_body(request)
+      if @body_attrs
+        request.body = serialize_attributes(attributes: @body_attrs).to_json
+      end
+    end
+
+    def assign_request_headers(request)
+      http_headers ||= default_headers
+      if request.method != "GET"
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs This is not the rule. `GET` can also require Authorization. Depends on the endpoint.

File: lib/hubble_observatory/request.rb [hunk 4]
```diff
@@ -41,32 +43,62 @@ def serialize_attributes(attributes:)
...
+      {"Content-Type" => "application/vnd.api+json"}
     end
 
     def host
       subdomain = ENV['HUBBLE_ENV'] == 'production' ? 'hubble' : 'rc-hubble'
       "#{subdomain}.fullscreen.net"
     end
+
+    # Equivalent to @params.transform_keys{|key| key.to_s.camelize :lower}
+    def query
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs What is this (and the following method) for?

Comment (bruce-fs) [inline lib/hubble_observatory/request.rb]: @claudiofullscreen - for get query params

Comment (bruce-fs) [inline lib/hubble_observatory/request.rb]: Actually I don't need to camelize...that's a mistake I think

Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs  I mean the "camel" part

Comment (bruce-fs) [inline lib/hubble_observatory/request.rb]: But I don't need it

File: lib/hubble_observatory/fs_employee.rb (hunk 1 above)
Comment (bruce-fs) [inline lib/hubble_observatory/fs_employee.rb]: Updated the readme

Comment (bruce-fs) [inline lib/hubble_observatory/fs_employee.rb]: Updated the readme

File: README.md [hunk 5]
```diff
@@ -69,6 +69,14 @@ talent_account = HubbleObservatory::TalentAccount.new id: '123456'
 talent_account.update email: 'newuser@example.com'
 # => '123456'
 ```
+### FsEmployee#create
```
Comment (claudiofullscreen) [inline README.md]: @bruce-fs Still says `create`. I'm happy with `create` because that is happening, the token is just a part of it.

File: README.md [hunk 6]
```diff
@@ -69,6 +69,14 @@ talent_account = HubbleObservatory::TalentAccount.new id: '123456'
...
+
+Obtain the token of a fullscreen employee given the appropriate access token:
```
Comment (claudiofullscreen) [inline README.md]: @bruce-fs It's `Fullscreen`, not `fullscreen`. And can be removed.

File: lib/hubble_observatory/request.rb [hunk 7]
```diff
@@ -12,16 +12,17 @@ class Request
     def initialize(attrs: {})
       @request_type = attrs.fetch :request_type, :get
       @route = attrs.fetch :route, "talent-accounts"
+      @query_params = attrs.fetch :query_params, {}
       @body_attrs = attrs.fetch :body_attrs, nil
-      @error_message = attrs.fetch :error_message, ->(body) {"Error: #{body}"}
+      @request_format = attrs.fetch :request_format, :json
+      @auth_header = attrs.fetch :auth_header, false
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs `fetch… false` is not needed, `attrs[:auth_header]` is enough

File: lib/hubble_observatory/request.rb [hunk 8]
```diff
@@ -12,16 +12,17 @@ class Request
     def initialize(attrs: {})
       @request_type = attrs.fetch :request_type, :get
       @route = attrs.fetch :route, "talent-accounts"
+      @query_params = attrs.fetch :query_params, {}
       @body_attrs = attrs.fetch :body_attrs, nil
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs same here (although not part of this PR). `attrs[:body_attrs]` is the same.

File: lib/hubble_observatory/request.rb [hunk 9]
```diff
@@ -41,27 +42,48 @@ def serialize_attributes(attributes:)
...
+    end
+
+    def assign_request_body(request)
+      if @body_attrs
+        request.body = serialize_attributes(attributes: @body_attrs).to_json
+      end
+    end
+
+    def assign_request_headers(request)
+      http_headers ||= default_headers
```
Comment (claudiofullscreen) [inline lib/hubble_observatory/request.rb]: @bruce-fs why `||=`? is `http_headers` defined somewhere else?

File: lib/hubble_observatory/request.rb (hunk 8 above)
Comment (bruce-fs) [inline lib/hubble_observatory/request.rb]: ok will tackle in separate PR

File: lib/hubble_observatory/request.rb (hunk 9 above)
Comment (bruce-fs) [inline lib/hubble_observatory/request.rb]: it is not, was just changing things fast @claudiofullscreen

File: README.md (hunk 6 above)
Comment (bruce-fs) [inline README.md]: Removed

File: README.md (hunk 5 above)
Comment (bruce-fs) [inline README.md]: Ok left as create
